spec/lib/cli.spec.js

```diff
@@ -0,0 +1,166 @@
+"use strict";
+describe('The cli', () => {
+  let fs;
+  let path;
+  let mockfs;
+  let cli;
+  let Project;
+
+  let dir;
+  let aureliaProject;
+  beforeEach(() => {
+    fs = require('../../lib/file-system');
+    path = require('path');
+    dir = 'workspaces';
+    const fsConfig = {};
+    fsConfig[dir] = {};
+
+    mockfs = require('mock-fs');
+    mockfs(fsConfig);
+
+    cli = new (require('../../lib/cli').CLI)();
+    Project = require('../../lib/project').Project;
+
+    aureliaProject = 'aurelia_project';
+  });
+
+  describe('The _establishProject() function', () => {
+    let project;
+    let establish;
+
+    beforeEach(() => {
+      project = {};
+      spyOn(Project, 'establish').and.returnValue(project);
+    });
+
+    it('resolves to nothing', done => {
+      cli._establishProject({})
+        .then(project => {
+          expect(project).not.toBeDefined();
+        })
+        .catch(fail).then(done);
+    });
+
+    it('calls and resolves to Project.establish()', done => {
+      fs.mkdirp(path.join(process.cwd(), aureliaProject))
+        .then(() => cli._establishProject({
+          runningLocally: true
+        }))
+        .then(project => {
+          expect(Project.establish)
+            .toHaveBeenCalledWith(cli.ui, path.join(process.cwd()));
+          expect(project).toBe(project);
+        })
+        .catch(fail).then(done);
+    });
+
+    it('does not catch Project.establish()', () => {
+      pending('Does it even make sense to test this?');
```

Comment:
As it says. The test would refer to the use of project.establish, returning a promise. If it were catched here, it could not be observed with a .catch() later. However, I feel like the effort of testing for these kind of things might be over the top compared to the benefits.